### PR TITLE
CI: unittest reporting with pytest

### PR DESCRIPTION
**Pull request by jnewb1**
so we can see the time for each test

depends: https://github.com/commaai/openpilot/pull/29538

### CODE REVIEW

File: .github/workflows/selfdrive_tests.yaml [hunk 1]
```diff
@@ -284,10 +270,16 @@ jobs:
...
                         ./tools/cabana/tests/test_cabana && \
                         ./system/camerad/test/ae_gray_test && \
-                        ./selfdrive/test/process_replay/test_fuzzy.py && \
-                        coverage xml"
+                        ./selfdrive/test/process_replay/test_fuzzy.py"
     - name: "Upload coverage to Codecov"
       uses: codecov/codecov-action@v3
+    - name: Test Report
+      uses: dorny/test-reporter@v1
+      if: success() || failure()    # run this step even if previous step failed
```
Comment (sshane) [inline .github/workflows/selfdrive_tests.yaml:278]:
```suggestion
      if: true
```
does this work?

**Comment by jnewb1** [comment]
This test reporter isn't great, doesn't show time for each test. Will keep looking

**Comment by jnewb1** [comment]
Perhaps pytest-html

https://github.com/pytest-dev/pytest-html

**Comment by adeebshihadeh** [comment]
This should be easy to do on PC first. Everything that happens in CI should be easy to run or test locally.

**Pull request closed without merging**